internal/integration_test/post1_0/bulk-memory-operations/spec_test.go

```diff
@@ -0,0 +1,85 @@
+package bulk_memory_operations
+
+import (
+	"context"
+	_ "embed"
+	"testing"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/internal/testing/require"
+)
+
+// testCtx is an arbitrary, non-default context. Non-nil also prevents linter errors.
+var testCtx = context.WithValue(context.Background(), struct{}{}, "arbitrary")
+
+func TestBulkMemoryOperations_JIT(t *testing.T) {
+	if !wazero.JITSupported {
+		t.Skip()
+	}
+	testBulkMemoryOperations(t, wazero.NewRuntimeConfigJIT)
+}
+
+func TestBulkMemoryOperations_Interpreter(t *testing.T) {
+	testBulkMemoryOperations(t, wazero.NewRuntimeConfigInterpreter)
+}
+
+// bulkMemoryOperationsWasm was compiled from testdata/bulk_memory_operations.wat
+//go:embed testdata/bulk_memory_operations.wasm
+var bulkMemoryOperationsWasm []byte
+
+func testBulkMemoryOperations(t *testing.T, newRuntimeConfig func() *wazero.RuntimeConfig) {
+	t.Run("disabled", func(t *testing.T) {
+		// bulk-memory-operations is disabled by default.
+		r := wazero.NewRuntimeWithConfig(newRuntimeConfig())
+		_, err := r.InstantiateModuleFromCode(testCtx, bulkMemoryOperationsWasm)
+		require.Error(t, err)
+	})
+	t.Run("enabled", func(t *testing.T) {
+		r := wazero.NewRuntimeWithConfig(newRuntimeConfig().WithFeatureBulkMemoryOperations(true))
+
+		// Test the example logic.
+		t.Run("base case", func(t *testing.T) {
+			a, err := r.NewModuleBuilder("a").
+				ExportGlobalI32("global", 0). // global=1 ignores the "memory.init" on .start.
+				Instantiate(testCtx)
+			require.NoError(t, err)
+			defer a.Close(testCtx)
+
+			mod, err := r.InstantiateModuleFromCode(testCtx, bulkMemoryOperationsWasm)
+			require.NoError(t, err)
+			defer mod.Close(testCtx)
+
+			// The first segment is active, so we expect it to be readable.
+			bytes, ok := mod.Memory().Read(testCtx, 0, 5)
+			require.True(t, ok)
+			require.Equal(t, "hello", string(bytes))
+
+			// As the value of the global was zero, we don't expect "memory.init" to have copied the passive segment.
+			bytes, ok = mod.Memory().Read(testCtx, 16, 7)
+			require.True(t, ok)
+			require.Equal(t, make([]byte, 7), bytes)
+		})
+		t.Run("memory.init", func(t *testing.T) {
+			a, err := r.NewModuleBuilder("a").
+				ExportGlobalI32("global", 1). // global=1 triggers the "memory.init" on .start.
+				Instantiate(testCtx)
+			require.NoError(t, err)
+			defer a.Close(testCtx)
+
+			mod, err := r.InstantiateModuleFromCode(testCtx, bulkMemoryOperationsWasm)
+			require.NoError(t, err)
+			defer mod.Close(testCtx)
+
+			// The first segment is active, so we expect it to be readable.
+			bytes, ok := mod.Memory().Read(testCtx, 0, 5)
+			require.True(t, ok)
+			require.Equal(t, "hello", string(bytes))
+
+			// As the value of the global was zero, we don't expect "memory.init" to have copied the passive segment.
+			bytes, ok = mod.Memory().Read(testCtx, 16, 7)
+			require.True(t, ok)
+			// TODO: require.Equal(t, "goodbye", string(bytes))
```

Comment:
@mathetake FYI I left this until it is implemented

Reply:
oh? is this table related?

Reply:
everything memory.fill copy and init is ensured to be implemented so this should work

Reply:
```
(module $bulk-memory-operations
   (import "a" "global" (global i32))  ;; global 0
   (memory 1)
   (data (i32.const 0) "hello")   ;; data segment 0, is active so always copied
   (data "goodbye")               ;; data segment 1, is passive

   (func $start
     (if (global.get 0)

       ;; copy data segment 1 into memory 0 (the 0 is implicit)
       (memory.init 1
         (i32.const 16)    ;; target offset
         (i32.const 0)     ;; source offset
         (i32.const 7))    ;; length

       ;; The memory used by this segment is no longer needed, so this segment can
       ;; be dropped.
       (data.drop 1))
   )
 )
```

the if should trigger and invoke the memory.init thingy.. wasn't sure why that didn't run assumed something wasn't done, yet. Let me double-check the wasm generated includes it.

Reply:
yeah seems to

```
$ wasm2wat  bulk_memory_operations.wasm 
(module
  (type (;0;) (func))
  (import "a" "global" (global (;0;) i32))
  (func (;0;) (type 0)
    global.get 0
    if  ;; label = @1
      i32.const 16
      i32.const 0
      i32.const 7
      memory.init 1
    else
      data.drop 1
    end)
  (memory (;0;) 1)
  (data (;0;) (i32.const 0) "hello")
  (data (;1;) "goodbye"))
``

Reply:
not following, but this only contains ones implemented so should work

Reply:
feature overview forgot to make the start function start. wasn't a bug on our side except copying the bug :D